Lib/logging/handlers.py

```diff
@@ -827,11 +828,26 @@ def __init__(self, address=('localhost', SYSLOG_UDP_PORT),
             self.unixsocket = False
             if socktype is None:
                 socktype = socket.SOCK_DGRAM
-            self.socket = socket.socket(socket.AF_INET, socktype)
-            if socktype == socket.SOCK_STREAM:
-                self.socket.connect(address)
-            self.socktype = socktype
-        self.formatter = None
+            host, port = address
+            err = None
+            for res in socket.getaddrinfo(host, port, 0, socktype):
+                af, socktype, proto, cannonname, sa = res
+                sock = None
+                try:
+                    sock = socket.socket(af, socktype, proto)
+                    if socktype == socket.SOCK_STREAM:
+                        sock.connect(sa)
+                    self.socket = sock
+                    self.socktype = socktype
+                    return
+                except OSError as _:
+                    err = _
+                    if sock is not None:
+                        sock.close()
+            if err is not None:
+                raise err
+            else:
```

Comment:
Are you referring to my other comment? If so, I agree that `return` is equivalent to `break` _functionally_, but I prefer to use `break` because then in future code can be added after the `for` loop if needed. I generally prefer to avoid `return` in the middle of functions when it's easy to do.